src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java

```diff
@@ -266,6 +271,15 @@ public static Component convertComponent(final org.cyclonedx.model.Component cdx
             component.setChildren(children);
         }
 
+        final List<ComponentProperty> properties = convertToComponentProperties(cdxComponent.getProperties());
+        if (component.getId() == 0) {
+            component.setProperties(properties);
+        } else {
+            try (var qm = new QueryManager()) {
+                qm.synchronizeComponentProperties(component, properties);
+            }
+        }
```

Comment:
Synchronization should not happen in the `ModelConverter`, we should keep persistence logic out of it as much as possible. Conversion happens in line 189 already. 

The synchronization should happen in the `BomUploadProcessingTask`, in the same DB transaction as everything else.

Reply:
yes, it seems `qm.synchronizeComponentProperties` is also already happening in `processComponents` in `BomUploadProcessingTask`, I'll remove this part entirely.